From e19e26cfc6059086de871a32ba61ebb58a2db6a5 Mon Sep 17 00:00:00 2001
From: Dan Aloni <alonid@gmail.com>
Date: Wed, 14 Jun 2017 18:46:28 +0300
Subject: [PATCH 30/30] Revert "Add extension autoloading based on resources."

This reverts commit d3fc23b4a4ac48a1185985cad764477e951da688.

It appears to crash on my system. Found via 'git bisect'.
---
 src/init.C      |  9 ---------
 src/rxvt.h      |  6 +-----
 src/urxvt.pm    |  2 --
 src/xdefaults.C | 14 +++++++-------
 4 files changed, 8 insertions(+), 23 deletions(-)

diff --git a/src/init.C b/src/init.C
index 8e812005dd77..a6fa9cad417e 100644
--- a/src/init.C
+++ b/src/init.C
@@ -600,14 +600,6 @@ rxvt_term::init_vars ()
   set_option (Opt_buffered);
 }
 
-#if ENABLE_PERL
-static void
-rxvt_perl_parse_resource (rxvt_term *term, const char *k, const char *v)
-{
-  rxvt_perl.parse_resource (term, k, false, false, 0, v);
-}
-#endif
-
 /*----------------------------------------------------------------------*/
 const char **
 rxvt_term::init_resources (int argc, const char *const *argv)
@@ -655,7 +647,6 @@ rxvt_term::init_resources (int argc, const char *const *argv)
       || (rs[Rs_perl_eval] && *rs[Rs_perl_eval]))
     {
       rxvt_perl.init (this);
-      enumerate_resources (rxvt_perl_parse_resource);
       HOOK_INVOKE ((this, HOOK_INIT, DT_END));
     }
 #endif
diff --git a/src/rxvt.h b/src/rxvt.h
index 94ea6d9e3e30..465d94598c13 100644
--- a/src/rxvt.h
+++ b/src/rxvt.h
@@ -1487,11 +1487,7 @@ struct rxvt_term : zero_initialized, rxvt_vars, rxvt_screen
   int bind_action (const char *str, const char *arg);
   const char *x_resource (const char *name);
   void extract_resources ();
-  void enumerate_resources (void (*cb)(rxvt_term *, const char *, const char *), const char *name_p = 0, const char *class_p = 0);
-  void enumerate_keysym_resources (void (*cb)(rxvt_term *, const char *, const char *))
-  {
-    enumerate_resources (cb, "keysym", "Keysym");
-  }
+  void enumerate_keysym_resources (void (*cb)(rxvt_term *, const char *, const char *));
   void extract_keysym_resources ();
 };
 
diff --git a/src/urxvt.pm b/src/urxvt.pm
index 3dcf70201d85..c879019e8d7a 100644
--- a/src/urxvt.pm
+++ b/src/urxvt.pm
@@ -606,8 +606,6 @@ sub parse_resource {
 
          push @{ $term->{perl_ext_3} }, $v->[0];
 
-         return 1 unless $isarg;
-
          if ($v->[1] eq "boolean") {
             $term->put_option_db ($name, $flag ? "true" : "false");
             return 1;
diff --git a/src/xdefaults.C b/src/xdefaults.C
index 9b47bf2c4151..c31fcc68a3bc 100644
--- a/src/xdefaults.C
+++ b/src/xdefaults.C
@@ -635,7 +635,7 @@ rxvt_define_key (rxvt_term *term, const char *k, const char *v)
  *   value will be a string
  */
 static int
-rxvt_enumerate_helper (
+rxvt_keysym_enumerate_helper (
    XrmDatabase *database ecb_unused,
    XrmBindingList bindings ecb_unused,
    XrmQuarkList quarks,
@@ -850,7 +850,7 @@ rxvt_term::extract_resources ()
 }
 
 void
-rxvt_term::enumerate_resources (void (*cb)(rxvt_term *, const char *, const char *), const char *name_p, const char *class_p)
+rxvt_term::enumerate_keysym_resources (void (*cb)(rxvt_term *, const char *, const char *))
 {
   /*
    * [R5 or later]: enumerate the resource database
@@ -865,25 +865,25 @@ rxvt_term::enumerate_resources (void (*cb)(rxvt_term *, const char *, const char
   XrmName name_prefix[3];
   XrmClass class_prefix[3];
 
-  name_prefix[1] = name_p ? XrmStringToName (name_p) : NULLQUARK;
+  name_prefix[1] = XrmStringToName ("keysym");
   name_prefix[2] = NULLQUARK;
-  class_prefix[1] = class_p ? XrmStringToName (class_p) : NULLQUARK;
+  class_prefix[1] = XrmStringToName ("Keysym");
   class_prefix[2] = NULLQUARK;
 
 # ifdef RESFALLBACK
   name_prefix[0] = class_prefix[0] = XrmStringToName (RESFALLBACK);
   /* XXX: Need to check sizeof (rxvt_t) == sizeof (XPointer) */
   XrmEnumerateDatabase (database, name_prefix, class_prefix,
-                        XrmEnumOneLevel, rxvt_enumerate_helper, (XPointer)closure);
+                        XrmEnumOneLevel, rxvt_keysym_enumerate_helper, (XPointer)closure);
 # endif
 
   name_prefix[0] = class_prefix[0] = XrmStringToName (RESCLASS);
   XrmEnumerateDatabase (database, name_prefix, class_prefix,
-                        XrmEnumOneLevel, rxvt_enumerate_helper, (XPointer)closure);
+                        XrmEnumOneLevel, rxvt_keysym_enumerate_helper, (XPointer)closure);
 
   name_prefix[0] = class_prefix[0] = XrmStringToName (rs[Rs_name]);
   XrmEnumerateDatabase (database, name_prefix, class_prefix,
-                        XrmEnumOneLevel, rxvt_enumerate_helper, (XPointer)closure);
+                        XrmEnumOneLevel, rxvt_keysym_enumerate_helper, (XPointer)closure);
 #endif
 }
 
-- 
2.9.4

